### ISSUE TITLE
duckpan installation fails because of Unicode::LineBreak

### ISSUE THREAD

**Issue by iambibhas** [duckduckgo/p5-app-duckpan#270]
This is the content of build.log - 

```
cpanm (App::cpanminus) 1.7039 on perl 5.018002 built for x86_64-linux-gnu-thread-multi
Work directory is /home/bibhas/.cpanm/work/1441465489.17003
You have make /usr/bin/make
You have LWP 6.13
You have /bin/tar: tar (GNU tar) 1.27.1
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching App::DuckPAN () on cpanmetadb ...
--> Working on App::DuckPAN
Fetching http://www.cpan.org/authors/id/M/MO/MOOLLAZA/App-DuckPAN-0.190.tar.gz
-> OK
Unpacking App-DuckPAN-0.190.tar.gz
Entering App-DuckPAN-0.190
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.06)
Checking if you have File::ShareDir::Install 0.06 ... Yes (0.10)
Configuring App-DuckPAN-0.190
Running Makefile.PL
Warning: prerequisite MooX::Options 0 not found.
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for App::DuckPAN
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Moo 0 ... Yes (2.000002)
Checking if you have MooX 0 ... Yes (0.101)
Checking if you have LWP::Simple 0 ... Yes (6.13)
Checking if you have Test::More 0 ... Yes (1.001014)
Checking if you have Encode 0 ... Yes (2.49)
Checking if you have Test::Script::Run 0 ... Yes (0.08)
Checking if you have Carp 0 ... Yes (1.29)
Checking if you have Plack::MIME 0 ... Yes (undef)
Checking if you have Time::HiRes 0 ... Yes (1.9725)
Checking if you have version 0 ... Yes (0.9902)
Checking if you have HTTP::Request::Common 0 ... Yes (6.10)
Checking if you have Getopt::Long 0 ... Yes (2.47)
Checking if you have Plack::Request 0 ... Yes (1.0037)
Checking if you have Email::Valid 0 ... Yes (1.192)
Checking if you have File::Temp 0 ... Yes (0.23)
Checking if you have Perl::Version 0 ... Yes (1.013)
Checking if you have strict 0 ... Yes (1.07)
Checking if you have HTML::Element 0 ... Yes (5.03)
Checking if you have Plack::Runner 0 ... Yes (undef)
Checking if you have Test::LoadAllModules 0 ... Yes (0.022)
Checking if you have Data::Dumper 0 ... Yes (2.145)
Checking if you have Class::Load 0 ... Yes (0.23)
Checking if you have File::Which 0 ... Yes (1.19)
Checking if you have File::ShareDir::ProjectDistDir 0 ... Yes (1.000008)
Checking if you have Term::ANSIColor 0 ... Yes (4.02)
Checking if you have MooX::Cmd 0 ... Yes (0.015)
Checking if you have HTTP::Status 0 ... Yes (6.10)
Checking if you have Module::Pluggable::Object 0 ... Yes (5.1)
Checking if you have Config::INI 0 ... Yes (0.025)
Checking if you have Term::ReadLine 0 ... Yes (1.12)
Checking if you have Dist::Zilla::Util 0 ... Yes (5.039)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.06)
Checking if you have HTTP::Request 0 ... Yes (6.10)
Checking if you have feature 0 ... Yes (1.32)
Checking if you have Text::Xslate 0 ... Yes (3.3.7)
Checking if you have lib 0 ... Yes (0.63)
Checking if you have POE 0 ... Yes (1.367)
Checking if you have LWP::UserAgent 0 ... Yes (6.13)
Checking if you have Term::UI 0 ... Yes (0.42)
Checking if you have Data::Printer 0 ... Yes (0.36)
Checking if you have JSON 0 ... Yes (2.90)
Checking if you have HTML::TreeBuilder 0 ... Yes (5.03)
Checking if you have Term::ProgressBar 0 ... Yes (2.17)
Checking if you have Path::Tiny 0 ... Yes (0.072)
Checking if you have Module::Data 0 ... Yes (0.007)
Checking if you have List::MoreUtils 0 ... Yes (0.413)
Checking if you have Plack::Handler::Starman 0 ... Yes (undef)
Checking if you have MooX::Options 0 ... No
Checking if you have URI::Escape 0 ... Yes (3.31)
Checking if you have Pod::Usage 0 ... Yes (1.61)
Checking if you have POE::Wheel::ReadLine 0 ... Yes (1.367)
Checking if you have Try::Tiny 0 ... Yes (0.22)
Checking if you have HTML::Entities 0 ... Yes (3.69)
Checking if you have Parse::CPAN::Packages::Fast 0 ... Yes (0.08)
Checking if you have File::Find::Rule 0 ... Yes (0.33)
Checking if you have List::Util 0 ... Yes (1.42)
Checking if you have Moo::Role 0 ... Yes (2.000002)
Checking if you have File::HomeDir 0 ... Yes (1.00)
Checking if you have File::Copy 0 ... Yes (2.26)
Checking if you have Config::INI::Reader 0 ... Yes (0.025)
Checking if you have warnings 0 ... Yes (1.18)
Checking if you have Filesys::Notify::Simple 0 ... Yes (0.12)
Checking if you have Config::INI::Writer 0 ... Yes (0.025)
Checking if you have Plack::Response 0 ... Yes (1.0037)
Checking if you have open 0 ... Yes (1.10)
==> Found dependencies: MooX::Options
Searching MooX::Options (0) on cpanmetadb ...
--> Working on MooX::Options
Fetching http://www.cpan.org/authors/id/C/CE/CELOGEEK/MooX-Options-4.018.tar.gz
-> OK
Unpacking MooX-Options-4.018.tar.gz
Entering MooX-Options-4.018
Checking configure dependencies from META.json
Checking if you have ExtUtils::Install 1.46 ... Yes (1.59)
Checking if you have Module::Build 0.4211 ... Yes (0.4214)
Configuring MooX-Options-4.018
Running Build.PL
Checking prerequisites...
  requires:
    !  Text::LineFold is not installed
  recommends:
    *  Term::Size::Any is not installed

ERRORS/WARNINGS FOUND IN PREREQUISITES.  You may wish to install the versions
of the modules indicated above before proceeding with this installation

Created MYMETA.yml and MYMETA.json
Creating new 'Build' script for 'MooX-Options' version '4.018'
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Text::LineFold 0 ... No
Checking if you have parent 0 ... Yes (0.225)
Checking if you have lib 0 ... Yes (0.63)
Checking if you have Carp 0 ... Yes (1.29)
Checking if you have Moo 1.003001 ... Yes (2.000002)
Checking if you have Test::More 0.94 ... Yes (1.001014)
Checking if you have feature 0 ... Yes (1.32)
Checking if you have IO::Handle 0 ... Yes (1.34)
Checking if you have File::Spec 0 ... Yes (3.40)
Checking if you have Test::Trap 0 ... Yes (0.3.2)
Checking if you have Getopt::Long::Descriptive 0.099 ... Yes (0.099)
Checking if you have Data::Record 0 ... Yes (0.02)
Checking if you have JSON 0 ... Yes (2.90)
Checking if you have Getopt::Long 2.43 ... Yes (2.47)
Checking if you have FindBin 0 ... Yes (1.51)
Checking if you have Try::Tiny 0 ... Yes (0.22)
Checking if you have overload 0 ... Yes (1.22)
Checking if you have Capture::Tiny 0 ... Yes (0.30)
Checking if you have strict 0 ... Yes (1.07)
Checking if you have English 0 ... Yes (1.06_01)
Checking if you have Pod::Usage 0 ... Yes (1.61)
Checking if you have Test::Requires 0 ... Yes (0.10)
Checking if you have MRO::Compat 0 ... Yes (0.12)
Checking if you have MooX::ConfigFromFile 0 ... Yes (0.007)
Checking if you have Role::Tiny::With 0 ... Yes (2.000001)
Checking if you have Regexp::Common 0 ... Yes (2013031301)
Checking if you have namespace::clean 0 ... Yes (0.25)
Checking if you have Scalar::Util 0 ... Yes (1.42)
Checking if you have Path::Class 0.32 ... Yes (0.35)
Checking if you have IPC::Open3 0 ... Yes (1.13)
Checking if you have Module::Metadata 1.000019 ... Yes (1.000027)
Checking if you have warnings 0 ... Yes (1.18)
Checking if you have Module::Build 0.4211 ... Yes (0.4214)
==> Found dependencies: Text::LineFold
Searching Text::LineFold (0) on cpanmetadb ...
--> Working on Text::LineFold
Fetching http://www.cpan.org/authors/id/N/NE/NEZUMI/Unicode-LineBreak-2015.07.16.tar.gz
-> OK
Unpacking Unicode-LineBreak-2015.07.16.tar.gz
Entering Unicode-LineBreak-2015.07.16
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.06)
Configuring Unicode-LineBreak-v2015.07.16
Running Makefile.PL
Use bundled sombok 2.4.0 with Unicode 8.0.0
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for sombok
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for Unicode::LineBreak
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Encode 1.98 ... Yes (2.49)
Checking if you have ExtUtils::MakeMaker 6.26 ... Yes (7.06)
Checking if you have MIME::Charset v1.6.2 ... Yes (1.012)
Checking if you have Test::More 0.45 ... Yes (1.001014)
Building and testing Unicode-LineBreak-v2015.07.16
cp lib/Unicode/LineBreak/Defaults.pm.sample blib/lib/Unicode/LineBreak/Defaults.pm.sample
cp lib/POD2/JA/Unicode/LineBreak.pod blib/lib/POD2/JA/Unicode/LineBreak.pod
cp lib/Unicode/LineBreak/Constants.pm blib/lib/Unicode/LineBreak/Constants.pm
cp lib/Unicode/LineBreak.pm blib/lib/Unicode/LineBreak.pm
cp lib/POD2/JA/Text/LineFold.pod blib/lib/POD2/JA/Text/LineFold.pod
cp lib/Unicode/LineBreak.pod blib/lib/Unicode/LineBreak.pod
cp lib/POD2/JA/Unicode/GCString.pod blib/lib/POD2/JA/Unicode/GCString.pod
cp lib/Text/LineFold.pm blib/lib/Text/LineFold.pm
cp lib/Unicode/GCString.pod blib/lib/Unicode/GCString.pod
cp lib/Unicode/GCString.pm blib/lib/Unicode/GCString.pm
cd sombok && make all
make[1]: Entering directory `/home/bibhas/.cpanm/work/1441465489.17003/Unicode-LineBreak-2015.07.16/sombok'
cd lib && cc -c  "-I/usr/lib/perl/5.18/CORE" "-I/home/bibhas/.cpanm/work/1441465489.17003/Unicode-LineBreak-2015.07.16/sombok/include"  -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fstack-protector -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"\" -DXS_VERSION=\"\" -fPIC "-I/usr/lib/perl/5.18/CORE"   ../lib/8.0.0.c -o lib/8.0.0.o
Assembler messages:
Fatal error: can't create lib/8.0.0.o: No such file or directory
make[1]: *** [lib/8.0.0.o] Error 2
make[1]: Leaving directory `/home/bibhas/.cpanm/work/1441465489.17003/Unicode-LineBreak-2015.07.16/sombok'
make: *** [sombok/libsombok.a] Error 2
-> FAIL Installing Text::LineFold failed. See /home/bibhas/.cpanm/work/1441465489.17003/build.log for details. Retry with --force to force install it.
-> FAIL Installing the dependencies failed: Module 'Text::LineFold' is not installed
-> FAIL Bailing out the installation for MooX-Options-4.018.
-> FAIL Installing the dependencies failed: Module 'MooX::Options' is not installed
-> FAIL Bailing out the installation for App-DuckPAN-0.190.
```

`--force` doesn't work. Any idea?

**Comment by mintsoft** [comment]
@iambibhas which O/S are you installing on?

**Comment by iambibhas** [comment]
@mintsoft this is elementary, which is Ubuntu 14.04 actually.

**Comment by mintsoft** [comment]
@iambibhas I'd try putting the system package on for the moox:
* http://packages.ubuntu.com/trusty/libmoox-options-perl

**Comment by iambibhas** [comment]
@mintsoft I think that did the trick! Thanks! I'll update once the install finishes. :) Maybe we can add this to the doc too? As a dependency.

**Comment by mintsoft** [comment]
@iambibhas *technically* it shouldn't be required as it should be able to resolve it all from cpan & duckpan together. However, on Debian based systems things are never that simple!

**Comment by iambibhas** [comment]
@mintsoft if I want to leave this debian based OS, which OS would you suggest?

**Comment by mintsoft** [comment]
@iambibhas I actually use Debian personally! It's stability and excellent packaging more than makes  up for a few issues with cpan, especially as they're usually resolvable with either installing a package or  alternatively for completely separate installations, using perlbrew, in my opinion. I've found that all distros have weird problems with `cpan`/`npm`/`pip`/`gem`/`pecl`/`pear` etc when they compete with officially packaged modules. 

So in short, I sadly have no recommendation I guess!

**Comment by iambibhas** [comment]
Ok. It finished installing. 2 system packages I had to install separately were `libmoox-options-perl` and `libmpfr-dev`. Thanks @mintsoft.

**Comment by iambibhas** [comment]
Just for reference, I had to install those above 2 packages on Linux Mint too, to get things working.

**Comment by moollaza** [comment]
FWIW this is related to a bug recently introduce in ExtUtils-MakeMaker AFAICT: https://rt.cpan.org/Public/Bug/Display.html?id=106840

**Comment by iambibhas** [comment]
We can probably open this issue until it's fixed on OSX.
